src/arraymancer/tensor/private/p_display.nim

```diff
@@ -215,4 +287,136 @@ proc prettyImpl*[T](t: Tensor[T],
+  new_shape.delete(0)
+  result = t.reshape(new_shape)
+
+proc dispTensorAsSingleLineArrayImp[T](t: Tensor[T],
+                    precision = -1,
+                    specifier: static string = "",
+                    indentSpacing = 0,
+                    sep = ", ", rowSep = ""
+                    ): string =
+  ## Implementation of the "array-style" tensor printing
+  result = "["
+  if t.rank <= 1:
+    result = disp1dAsArray(t, sep = sep, precision = precision, specifier = specifier)
+  else:
+    var n = 0
+    for ax in axis(t, 0):
+      var axRepr = dispTensorAsSingleLineArrayImp(ax.squeezeTopDimension(),
+        precision = precision,
+        specifier = specifier,
+        indentSpacing = indentSpacing,
+        sep = sep, rowSep = rowSep)
+      result &= axRepr
+      n += 1
+      if n < t.shape[0]:
+        result &= sep & rowSep
+    result &= "]"
+
+proc dispTensorAsSingleLineArray*[T](t: Tensor[T],
+                    precision = -1,
+                    indentSpacing = 0,
+                    specifier: static string = "",
+                    sep = ", ", rowSep = "",
+                    showHeader = true
+                    ): string =
+  ## Display a tensor as a single line "array"
+  # Remove the non-standard specifier flags (n, a and s)
+  const specifier = removeTensorFormatSpecifiers(specifier)
+  if showHeader:
+    result = t.compactTensorDescription & ":" & rowSep
+  result &= dispTensorAsSingleLineArrayImp(t, precision, specifier = specifier, rowSep = rowSep)
+  if t.storage.isNil:
+    # Return a useful message for uninit'd tensors instead of crashing
+    # Note that this should only happen when displaying tensors created
+    # by just declaring their type (e.g. `var t: Tensor[int]`), given that
+    # even tensors created by calling `newTensorUninit` have their storage
+    # initialized (with garbage)
+    result &= " (uninitialized)"
+
+proc indentTensorReprRows(s: string, indent: int): string =
+  ## Indent the lines of a multi-line "array-style" tensor representation
+  ## so that the right-most opening braces align vertically
+  if indent <= 0:
+    return s
+  for line in s.splitLines():
+    var numBrackets = 0
+    for c in line:
+      if c != '[':
+        break
+      numBrackets += 1
+    result &= line.indent(indent - numBrackets) & "\n"
+
+proc dispTensorAsArray*[T](t: Tensor[T],
+                    precision = -1,
+                    specifier: static string = "",
+                    showHeader = true): string =
+  ## Display a tensor as a multi-line "array"
+  result = t.dispTensorAsSingleLineArray(
+    precision = precision, specifier = specifier, rowSep="\n", showHeader = false)
+  result = indentTensorReprRows(result, t.rank).strip(leading=false)
+  if showHeader:
+    result = t.compactTensorDescription() & ":\n" & result
+
+proc prettyImpl*[T](
+    t: Tensor[T], precision: int, specifier: static string): string =
+  ## Non public implementation of the pretty function
```

Comment:
```suggestion
  ## Public implementation of the pretty function
```
?

Reply:
Users are not expected to use the `prettyImpl` function. It is "public" in the sense that it is exported from `p_disp.nim`, but `p_disp.nim` itself is not exported so in the end it should be non-public?